core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -36,15 +37,33 @@ import org.apache.kafka.common.errors.{GroupAuthorizationException, TopicAuthori
 import org.apache.kafka.common.resource._
 import org.apache.kafka.common.resource.ResourceType._
 import org.apache.kafka.common.resource.PatternType.{LITERAL, PREFIXED}
-import org.apache.kafka.common.security.auth.KafkaPrincipal
+import org.apache.kafka.common.security.auth._
+import org.apache.kafka.common.security.authenticator.DefaultKafkaPrincipalBuilder
+import org.apache.kafka.common.config.internals.BrokerSecurityConfigs
 import org.apache.kafka.server.metrics.KafkaYammerMetrics
 import org.junit.jupiter.api.Assertions._
-import org.junit.jupiter.api.{AfterEach, BeforeEach, Test, TestInfo, Timeout}
+import org.junit.jupiter.api.{AfterEach, BeforeEach, TestInfo, Timeout}
 import org.junit.jupiter.params.ParameterizedTest
 import org.junit.jupiter.params.provider.ValueSource
 
 import scala.jdk.CollectionConverters._
 
+/**
+  * All broker to controller communications is authenticated as PLAINTEXT in these tests.
```

Comment:
For broker -> controller communication, we forward the authenticated principal (`User:server`) on broker to controller nodes using forward request mechanism. 

we can include "User:ANONYMOUS" to `super.users` config to allow PLAINTEXT communication between  controller nodes communication.

Reply:
Okay, I fixed this. Thanks for the tip.